### ISSUE TITLE
Help Getting Started

### ISSUE THREAD

**Issue by rbrunyon** [matpalm/bnn#13]
Hi,

I was wondering if you could help me out with getting started on using this program.  If I am starting with new images, which parts should I delete from a fresh download of yours (i.e. should I delete the existing labels.db file and place my new pictures into 'training' 'test' and 'unlabelled' folders? I tried following your README file but I am having some issues making it work. Any assistance that you could provide would be greatly appreciated.

Thank you,

**Comment by NunoCapela17** [comment]
Hi,
I'm in the same situation - need some help over here. 
I've made some videos of a hive entrance and I would like to know if it's possible to analyse those videos on my computer using your program.
Also, I didn't get it, but the program distinguish between bees going in and going out of the hive or only counts their activity?
Cheers

**Comment by squeakus** [comment]
@rbrunyon can you be more specific where you are having problems? Delete the labels.db and then create a new one using the labelling GUI. If you want to skip that step you can download my dataset:
https://www.kaggle.com/jonathanbyrne/to-bee-or-not-to-bee

@NunoCapela17 you will need to extract the video into individual frames for labelling and analysis.

All the code does is identify the positions of bees in the image, tracking between frames, directions etc will require you to code something up. Here is a tutorial explaining an approach to doing this:
https://www.pyimagesearch.com/2018/07/23/simple-object-tracking-with-opencv/

**Comment by rbrunyon** [comment]
@squeakus Absolutely.  No problem.  I begin by deleting the 'labels' folder and 'labels.db' file.  Then I place one different photo in each of the 'training', 'test', and 'unlabelled' folders.  After that, I have been able to successfully run 'label_ui.py' for the photos in both the 'training' and 'test' folders. When I run 'materialise_label_db.py' I get the following warnings but it seems to still work:

/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: compiletime version 3.4 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.5
  return f(*args, **kwds)
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: builtins.type size changed, may indicate binary incompatibility. Expected 432, got 412
  return f(*args, **kwds)
Namespace(directory='sample_data/labels/', height=1024, label_db='sample_data/labels.db', label_rescale=0.5, width=768)

Next, I proceed to run 'data.py' for the photos in both the 'training' and 'test' folders.  When I run this I get the following warnings as well (however it seems to work as well and produces the 'test_00X_XXX.png' photos like it should):

/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: compiletime version 3.4 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.5
  return f(*args, **kwds)
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: builtins.type size changed, may indicate binary incompatibility. Expected 432, got 412
  return f(*args, **kwds)
Namespace(batch_size=4, distort=False, height=1024, image_dir='sample_data/training/', label_dir='sample_data/labels/', label_rescale=0.5, patch_width_height=None, rotate=False, width=768)

 Then I try running 'train.py' with the following inputs:

./train.py \
--run r12 \
--steps 300 \
--train-steps 50 \
--train-image-dir sample_data/training/ \
--test-image-dir sample_data/test/ \
--label-dir sample_data/labels/ \
--width 768 --height 1024

and I get the following result:

/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: compiletime version 3.4 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.5
  return f(*args, **kwds)
/usr/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: builtins.type size changed, may indicate binary incompatibility. Expected 432, got 412
  return f(*args, **kwds)
opts Namespace(base_filter_size=8, batch_size=32, flip_left_right=False, height=1024, label_dir='sample_data/labels/', learning_rate=0.001, no_use_batch_norm=False, no_use_skip_connections=False, patch_width_height=None, pos_weight=1.0, random_rotate=False, run='r12', secs=None, steps=300, test_image_dir='sample_data/test/', train_image_dir='sample_data/training/', train_steps=50, width=768)
len(rgb_filenames) 1 CACHE
patch train model...
input      (?, 1024, 768, 3)    #2359296
e1         (?, 512, 384, 8)     #1572864
e2         (?, 256, 192, 16)    #786432
e3         (?, 128, 96, 32)     #393216
e4         (?, 64, 48, 64)      #196608
d1         (?, 128, 96, 32)     #393216
d1+e3      (?, 128, 96, 64)     #786432
d2         (?, 256, 192, 16)    #786432
d2+e2      (?, 256, 192, 32)    #1572864
d3         (?, 512, 384, 8)     #1572864
d3+e1      (?, 512, 384, 16)    #3145728
logits     (?, 512, 384, 1)     #196608
full res test model...
input      (?, 1024, 768, 3)    #2359296
e1         (?, 512, 384, 8)     #1572864
e2         (?, 256, 192, 16)    #786432
e3         (?, 128, 96, 32)     #393216
e4         (?, 64, 48, 64)      #196608
d1         (?, 128, 96, 32)     #393216
d1+e3      (?, 128, 96, 64)     #786432
d2         (?, 256, 192, 16)    #786432
d2+e2      (?, 256, 192, 32)    #1572864
d3         (?, 512, 384, 8)     #1572864
d3+e1      (?, 512, 384, 16)    #3145728
logits     (?, 512, 384, 1)     #196608
Killed

Sometimes instead of 'Killed' at the end, it will produce ''Segmentation Fault"

Any assistance that you could provide would be greatly appreciated.

Thank you

**Comment by matpalm** [comment]
for completeness i've added ba0a99c which includes an update to the README to reference `run_sample_training_pipeline.sh` which should work end to end. 

@rbrunyon  the compiletime version 3.4 / 3.5 is to do with the install you have of tensorflow. i don't think there's anything specific to this code that triggers it. i.e. i imagine you see it all the time? see https://github.com/tensorflow/tensorflow/issues/14182 the killed / seg fault is interesting because that has to be coming from tf lib code. what resolution are your images? can you share one?

@NunoCapela17 this code as is just supports a binary decision per pixel; `P(bee)` i did some initial experiments with `{no_bee, bee_facing_left, bee_facing_right}` and had reasonable results (but never finished it). the change to the model is trivial; we just change the [final output](https://github.com/matpalm/bnn/blob/master/model.py#L83
) to be 3 values instead of 1 and switch the loss from `weighted_cross_entropy_with_logits` to `softmax_cross_entropy_with_logits`. the much bigger change is all the plumbing; from the labelling tool to the data pipeline etc etc that need to now support a categorical version of the label (H, W, 3) rather than just the binary one (H, W, 1). generalising the code wouldn't be a bad thing to do though since the binary case (as the code is now) is exactly equivalent to a 2 class classification, and once you support 2, you can support N, e.g. `{nothing, bee_left, bee_right, butterfly, hornet, whatever, ... }`

**Comment by rbrunyon** [comment]
@matpalm Thank you for the help.  I really appreciate it. I was skeptical of my tensorflow install.  I will look into the link that you provided.  In the meantime, the photos that I am using are a few of the ones that you provided with the download on GitHub.  I have attached them below for easy reference: 
![20180213_165539](https://user-images.githubusercontent.com/43455835/49120326-fe2dd980-f260-11e8-9f4f-809f9ea03deb.jpg)
![20180213_122022](https://user-images.githubusercontent.com/43455835/49120341-04bc5100-f261-11e8-80fd-8494cb02c463.jpg)
![20180210_121423](https://user-images.githubusercontent.com/43455835/49120343-0b4ac880-f261-11e8-9df1-dc3da1fbb3c3.jpg)

**Comment by matpalm** [comment]
Does the run_sample_training_pipeline.sh script run ok?

**Comment by rbrunyon** [comment]
@matpalm I tried executing 'run_sample_training_pipeline.sh' in the terminal and it doesn't seem to get past 'data.py'

**Comment by matpalm** [comment]
I just ran it ok. TensorFlow is fine for you otherwise? (Ie other apps)

**Comment by rbrunyon** [comment]
I haven't really used tensorflow in other apps... Is there a way to tell if tensorflow is functioning properly?

**Comment by matpalm** [comment]
i don't know of one; but given the seg fault running code that i know works for others + those warnings i'd say it's a good chance. i'd go for a reinstall.
